mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -252,7 +250,7 @@ class LinearRegression @Since("1.3.0") (@Since("1.3.0") override val uid: String
           model,
           Array(0D),
           Array(0D))
-        return copyValues(model.setSummary(trainingSummary))
+        return model.setSummary(trainingSummary)
```

Comment:
@MLnick I added test case for this scenario and updated other test cases to ensure coping prediction column(and other params) correct in all situations.